zilliqa/src/api/zil.rs

```diff
@@ -378,3 +380,34 @@ fn get_tx_block(
 
     Ok(Some(block))
 }
+
+fn get_smart_contracts(params: Params, node: &Arc<Mutex<Node>>) -> Result<Vec<SmartContract>> {
+    let address: H160 = params.one()?;
+
+    let nonce = node
+        .lock()
+        .unwrap()
+        .get_account(address, BlockNumber::Latest)?
+        .nonce;
+
+    let mut contracts = vec![];
+
+    for i in 0..nonce {
```

Comment:
This feels like the kind of thing a hypothetical indexer would cache in its database rather than looping through every integer up to the nonce to check the address, on every RPC call.

The fact that it's a required RPC for zilliqa nodes (rather than optional functionality that could be enabled only for voluntary indexerers) makes me hesistant to say that we should therefore force every node to process and cache this in the database, though. I suppose it'll probably be fine (especially if we have a way to charge for API access anyway).

Reply:
Yeah fair point. I copied the implementation from ZQ1. I am quite sure we will add configuration to enable/disable specific APIs soon and this might be one that is only necessary for nodes serving devex. We could also reasonably limit the number of returned transactions to bound the work here.